src/server/utils/bucket_diff.js

```diff
@@ -454,5 +470,16 @@ class BucketDiff {
         return target_pos;
     }
 
+    /**
+     * @param {any[]} bucket_obj
+     */
+    _get_latest_delete_marker(bucket_obj) {
+        // Extract the latest delete marker from the bucket
+        const latest_delete_marker = bucket_obj.reduce(function(prev, current) {
+            return (new Date(prev.LastModified) > new Date(current.LastModified)) ? prev : current;
+        });
+        return latest_delete_marker;
+    }
```

Comment:
I think we do not need that.

Reply:
You mean we need not to fetch latest delete marker in different function? We should do it in the code itself instead of using different function, right?

Reply:
@liranmauda as per this aws s3 [api](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteMarkerEntry.html) 

Delete markers only contains fields - `IsLatest`, `Key`, `LastModified`, `VersionID`, `Owner`. And it doesn't contain the `etag` field, so to identify the keys on the basis of etags similar to version replication is not possible in delete replication. We cannot do it even with the `VersionID` as it can contain two different objects in different buckets.
This code here will get the diff keys from delete markers on the basis of `LastModified` and `IsLatest`. So, with this we are only able to replicate the latest delete markers.